modules/core/src/main/java/org/apache/ignite/internal/causality/OutdatedTokenException.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.causality;
+
+import org.apache.ignite.lang.IgniteInternalCheckedException;
+import org.apache.ignite.lang.IgniteStringFormatter;
+
+/**
+ * This exception is thrown when getting a value from {@link VersionedValue#get(long)} with a token outnound of stored history
```

Comment:
I would like to suggest slightly different wording:
```
This exception is thrown when {@link VersionedValue#get(long)} is called with an outdated token (this means that the history size of VersionedValue is not enough in order to get a value related to the token).
```

Reply:
I agree.